timeout@narutrey/files/timeout@narutrey/5.4/applet.js

```diff
@@ -0,0 +1,395 @@
+    },
+
+    open: function(isShowButtonPostpone, timestamp) {
+        let isOpened = ModalDialog.ModalDialog.prototype.open.call(this, timestamp);
+        if (!isOpened) return false;
+
+        if (isShowButtonPostpone) {
+            this.setButtons([
+                {
+                    label: _("Postpone"),
+                    action: Lang.bind(this, this.close),
+                    key: Clutter.Return
+                }
+            ]);
+        } else {
+            this._buttonLayout.destroy_all_children();
+        }
+        return true;
+    },
+}
+
+function Menu(launcher, orientation, isEnable) {
+    this._init(launcher, orientation, isEnable);
+}
+
+Menu.prototype = {
+    __proto__: Applet.AppletPopupMenu.prototype,
+
+    _menuManager:   null,
+    _switchEnable:  null,
+
+    _init: function(launcher, orientation, isEnable) {
+        Applet.AppletPopupMenu.prototype._init.call(this, launcher, orientation);
+
+        this._menuManager = new PopupMenu.PopupMenuManager(launcher);
+        //
+
+        this._switchEnable = new PopupMenu.PopupSwitchMenuItem(_("Enabled"));
+        this._switchEnable.setToggleState(isEnable);
+        this._switchEnable.connect('toggled', Lang.bind(this,
+            function() {
+                this.emit('toggled::enable');
+            }
+        ));
+        this.addMenuItem(this._switchEnable);
+
+        this.addAction(_("Reset timer"), Lang.bind(this,
+            function() {
+                this.emit('reset');
+            }
+        ));
+        this.addAction(_("Take a break"), Lang.bind(this,
+            function() {
+                this.emit('break');
+            }
+        ));
+    },
+
+    open: function(animate) {
+        this._menuManager.addMenu(this);
+        Applet.AppletPopupMenu.prototype.open.call(this, animate);
+    },
+
+    close: function(animate) {
+        if (this.isOpen) {
+            Applet.AppletPopupMenu.prototype.close.call(this, animate);
+            this._menuManager.removeMenu(this);
+        }
+    },
+
+    setEnable: function(isEnable) {
+        this._switchEnable.setToggleState(isEnable);
+    },
+}
+
+function formatSexagesimal(value) {
+    if (value < 0) value = -value;
+    value = Math.floor(value % 60);
+    if (value < 10) value = '0' + value;
+    return value;
+}
+
+function formatTime(seconds, isReturnSeconds, isShowHours) {
+    let s = formatSexagesimal(seconds);
+    let m = formatSexagesimal(seconds / 60);
+    let h = formatSexagesimal(seconds / (60 * 60));
+    let label;
+    if (isReturnSeconds) {
+        if (isShowHours) {
+            label = h + ':' + m + ':' + s;
+        } else {
+            label = m + ':' + s;
+        }
+    } else {
+        if (isShowHours) {
+            label = h + ':' + m;
+        } else {
+            m = parseInt(m, 10); // Drop leading zero
+            if (seconds % 60 > 0) ++m;
+            label = '' + m;
+        }
+    }
+    return label;
+}
```

Comment:
Take a look [here](https://github.com/linuxmint/cinnamon-spices-extensions/tree/master/mnemonic-5-4-window-menu%40mtwebster/files/mnemonic-5-4-window-menu%40mtwebster) for how to properly disable this past a certain point. You would need to create a 5.6 directory (assuming you still want 5.4 supported) with null functions. If you don't want this enabled for 5.4, then move the full code to 5.2 and the null functions to a 5.4 subdirectory.